okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/ExchangeFinder.kt

```diff
@@ -301,7 +270,6 @@ class ExchangeFinder(
   }
 
   fun trackFailure(e: IOException) {
-    nextRouteToTry = null
```

Comment:
I don’t think this made any sense. We were clearing out a stashed route when an unrelated route failed. For example, if we coalesced connections this would clear out the original route that we coalesced from. (This could be the cause of our difficult-to-reproduce `exhausted all routes` errors.)